pkg/cmd/project/item-add/item_add.go

```diff
@@ -0,0 +1,157 @@
+package itemadd
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/cli/cli/v2/pkg/cmdutil"
+
+	"github.com/cli/cli/v2/pkg/cmd/project/shared/format"
+	"github.com/cli/cli/v2/pkg/cmd/project/shared/queries"
+
+	"github.com/cli/cli/v2/internal/tableprinter"
+	"github.com/cli/go-gh/v2/pkg/api"
+	"github.com/shurcooL/githubv4"
+	"github.com/spf13/cobra"
+)
+
+type addItemOpts struct {
+	userOwner string
+	orgOwner  string
+	number    int
+	itemURL   string
+	projectID string
+	itemID    string
+	format    string
+}
+
+type addItemConfig struct {
+	tp     *tableprinter.TablePrinter
+	client *api.GraphQLClient
+	opts   addItemOpts
+}
+
+type addProjectItemMutation struct {
+	CreateProjectItem struct {
+		ProjectV2Item queries.ProjectItem `graphql:"item"`
+	} `graphql:"addProjectV2ItemById(input:$input)"`
+}
+
+func NewCmdAddItem(f *cmdutil.Factory, runF func(config addItemConfig) error) *cobra.Command {
+	opts := addItemOpts{}
+	addItemCmd := &cobra.Command{
+		Short: "Add a pull request or an issue to a project",
+		Use:   "item-add [number]",
+		Example: `
+# add an item to the current user's project 1
+gh project item-add 1 --user "@me" --url https://github.com/cli/go-gh/issues/1
+
+# add an item to user monalisa's project 1
+gh project item-add 1 --user monalisa --url https://github.com/cli/go-gh/issues/1
+
+# add an item to the org github's project 1
+gh project item-add 1 --org github --url https://github.com/cli/go-gh/issues/1
+
+# add --format=json to output in JSON format
+`,
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := queries.NewClient()
+			if err != nil {
+				return err
+			}
+
+			if len(args) == 1 {
+				opts.number, err = strconv.Atoi(args[0])
+				if err != nil {
+					return fmt.Errorf("invalid number: %v", args[0])
+				}
+			}
+
+			t := tableprinter.New(f.IOStreams)
+			config := addItemConfig{
+				tp:     t,
+				client: client,
+				opts:   opts,
+			}
+
+			// allow testing of the command without actually running it
+			if runF != nil {
+				return runF(config)
+			}
+			return runAddItem(config)
+		},
+	}
+
+	addItemCmd.Flags().StringVar(&opts.userOwner, "user", "", "Login of the user owner. Use \"@me\" for the current user.")
+	addItemCmd.Flags().StringVar(&opts.orgOwner, "org", "", "Login of the organization owner.")
+	addItemCmd.Flags().StringVar(&opts.itemURL, "url", "", "URL of the issue or pull request to add to the project. Note that the name of the owner is case sensitive, and will fail to find the item if it does not match. Must be of form https://github.com/OWNER/REPO/issues/NUMBER or https://github.com/OWNER/REPO/pull/NUMBER")
```

Comment:
The size of this flag's description raises alarm bells for me. Could it just be "URL of the issue or pull request to add to the project", with everything else being handled as an implementation detail, and the user not necessarily be aware of it?

Reply:
it could be - a user had opened an issue in the extension with this, because while https://github.com/cli/cli and https://github.com/CLI/cli will resolve in github.com, they won't both resolve in https://docs.github.com/en/graphql/reference/queries#resource

to avoid confusion I added a long disclaimer, if you feel it's more trouble than it's worth it can definitely be removed